src/_pytest/nodes.py

```diff
@@ -13,6 +14,7 @@
 tracebackcutdir = py.path.local(_pytest.__file__).dirpath()
 
 
+@lru_cache(maxsize=None)
```

Comment:
a cache of mutable return values is asking for trouble -- if this returned `tuple`s I'd be more comfortable

Reply:
Yeah, it would be safer. @Nepherhotep would you mind changing that so we can see how it fares?

Reply:
@nicoddemus, a simple converting response format from list to tuple caused 1500 errors. Most of them failed with the error "fixture testdir not found", some of them failed with repr error.
I need to dig into "testdir" fixture issue a bit deepr.

Reply:
I'm surprised... strange.

Reply:
Oh you missed the early return:

```python
    if nodeid == "":
        # If there is no root node at all, return an empty list so the caller's logic can remain sane
        return []
```

So sometimes it would return a tuple, others a list.

Reply:
Ah, my bad, thanks for pointing to that!